src/lib/utils/lossless_predicate_cast.hpp

```diff
@@ -0,0 +1,55 @@
+#include "lossless_cast.hpp"
+#include "types.hpp"
+
+namespace opossum {
+std::optional<float> next_float_towards(double value, double towards);
+
+// Even if lossless_cast is not successful, we might be able to do a cast if we know the predicate that is used.
+// As 3.1 has different float representations as a double and as a float (and neither is exact), lossless_cast will
+// fail. However, `float(a) > double(3.1)` can be reformulated as `float(a) >= float(3.099999904632568359375)`.
```

Comment:
It's supposed to be reformulated as the next-higher float, right?

Reply:
Good catch.